lldb/include/lldb/API/SBWatchpointOptions.h

```diff
@@ -0,0 +1,44 @@
+//===-- SBWatchpointOptions.h -----------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_API_SBWATCHPOINTOPTIONS_H
+#define LLDB_API_SBWATCHPOINTOPTIONS_H
+
+#include "lldb/API/SBDefines.h"
+
+class WatchpointOptionsImpl;
+
+namespace lldb {
+
+class LLDB_API SBWatchpointOptions {
+public:
+  SBWatchpointOptions();
+
+  SBWatchpointOptions(const lldb::SBWatchpointOptions &rhs);
+
+  ~SBWatchpointOptions();
+
+  const SBWatchpointOptions &operator=(const lldb::SBWatchpointOptions &rhs);
+
```

Comment:
add a header doc explaining this will stop any time memory is read from. Might want to mention if this works along with "modify" at all to only stop when reading a value and that value has changed?

Reply:
I don't think we can do read+modify, actually. I fixed a subtle bug in this latest commit in Watchpoint::WatchedValueReportable() where we know that a watched memory region may have been accessed.  If it's read+modify, we don't know if it's been read from, or written to (value mutating or not).  Unless we emulate the instruction that was executed, it could be a read, or a write with the same value, or it could be an access *near* the watched region which triggered the stop.  (this latter happening with large watchpoint support where watching a 32 byte region may actually need to watch 32 bytes, or on an AArch64 processor in streaming SVE mode when a write is *near* the watched region)